homeassistant/components/media_player/channels.py

```diff
@@ -0,0 +1,395 @@
+                response = requests.put(url, params, timeout=TIMEOUT)
+            elif method == "DELETE":
+                response = requests.delete(url, timeout=TIMEOUT)
+
+            if response:
+                return response.json()
+            else:
+                return {'status': 'error'}
+        except requests.exceptions.HTTPError:
+            return {'status': 'error'}
+        except requests.exceptions.Timeout:
+            return {'status': 'offline'}
+        except requests.exceptions.RequestException:
+            return {'status': 'offline'}
+
+    def _command(self, named_command):
+        """Make a request for a controlling command."""
+        return self._request('POST', '/api/' + named_command)
+
+    def status(self):
+        """Return the current state."""
+        return self._request('GET', '/api/status')
+
+    def favorite_channels(self):
+        """Return the favorite channels."""
+        response = self._request('GET', '/api/favorite_channels')
+        if "favorite_channels" in response:
+            return response["favorite_channels"]
+        else:
+            return []
+
+    def pause(self):
+        """Set playback to paused and returns the current state."""
+        return self._command('pause')
+
+    def resume(self):
+        """Set playback to play and returns the current state."""
+        return self._command('resume')
+
+    def stop(self):
+        """Set playback to stop and returns the current state."""
+        return self._command('stop')
+
+    def seek(self, seconds):
+        """Seek number of seconds."""
+        seconds = str(seconds or 0)
+        return self._command('seek/' + seconds)
+
+    def seek_forward(self):
+        """Seek forward."""
+        return self._command('seek_forward')
+
+    def seek_backward(self):
+        """Seek backward."""
+        return self._command('seek_backward')
+
+    def skip_forward(self):
+        """Skip forward to the next chapter mark."""
+        return self._command('skip_forward')
+
+    def skip_backward(self):
+        """Skip backward to the previous chapter mark."""
+        return self._command('skip_backward')
+
+    def toggle_muted(self):
+        """Mute and returns the current state."""
+        return self._command('toggle_mute')
+
+    def play_channel(self, channel_number):
+        """Set a channel to play and returns the current state."""
+        return self._request('POST', '/api/play/channel/' +
+                             str(channel_number))
+
+    def play_recording(self, recording_id):
+        """Set a recording to play and returns the current state."""
+        return self._request('POST', '/api/play/recording/' +
+                             str(recording_id))
+
+# pylint: disable=unused-argument, abstract-method
+# pylint: disable=too-many-instance-attributes
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Channels platform."""
+
+    device = ChannelsApp(
```

Comment:
Anyway you could validate the credentials after instantiating the object?

Reply:
I'm not sure what you mean. like, check for a response with the client or something? Do that before adding with `add_devices`?